internal/messenger/sms/sms.go

```diff
@@ -75,16 +77,40 @@ func (e *SMSSender) Push(m messenger.Message) error {
 		EnableTrace().
 		Post(srv.Host)
 
-	fmt.Println("Response Info:")
 	fmt.Println("  Error      :", err)
 	fmt.Println("  Status Code:", resp.StatusCode())
 	fmt.Println("  Status     :", resp.Status())
-	fmt.Println("  Proto      :", resp.Proto())
-	fmt.Println("  Time       :", resp.Time())
-	fmt.Println("  Received At:", resp.ReceivedAt())
 	fmt.Println("  Body       :\n", resp)
 	fmt.Println()
-	return err
+
+	var response Response
+
+	json.Unmarshal([]byte(resp.Body()), &response)
+	messageId := response.SMSMessageData.Recipients[0].MessageID
```

Comment:
@rbrn not sure why you are getting the first element in the array here, they are returning the status of all the telephone numbers that was sent to. So we need to insert each one of them.